src/query/storage/m3/storage.go

```diff
@@ -327,12 +310,17 @@ func (s *m3storage) fetchCompressed(
 				)
 			}
 
+			if err != nil {
+				result.ReportError(err)
```

Comment:
Should the below not happen if err != nil?

Reply:
Nice catch